mitosheet/mitosheet/public/v3/sheet_functions/number_functions.py

```diff
@@ -55,10 +82,182 @@ def AVG(*argv: NumberInputType) -> NumberFunctionReturnType:
 
     return sum_for_avg / num_entries if num_entries is not 0 else 0
 
+@cast_values_in_arg_to_type('s1', 'number')
+@cast_values_in_arg_to_type('s2', 'number')
+@handle_sheet_function_errors
+def CORR(s1: NumberRestrictedInputType, s2: NumberRestrictedInputType) -> NumberFunctionReturnType:
+    """
+    {
+        "function": "CORR",
+        "description": "Computes the correlation between two series, excluding missing values.",
+        "search_terms": ["corr", "correlation", "r^2"],
+        "examples": [
+            "=CORR(A, B)",
+            "=CORR(B, A)"
+        ],
+        "syntax": "CORR(series_one, series_two)",
+        "syntax_elements": [{
+                "element": "series_one",
+                "description": "The number series to convert to calculate the correlation."
+            }, {
+                "element": "series_two",
+                "description": "The number series to convert to calculate the correlation."
+            }
+        ]
+    }
+    """
+    if isinstance(s1, int) or isinstance(s1, float) or isinstance(s2, int) or isinstance(s2, float):
```

Comment:
We could use the `is_primitive_value` helper function for this check. Or create a `is_primitive_number` function. 

We could use this in other functions like, `FLOAT`, `LOG`, etc.

Reply:
We actually can't, because then we loose type information. This is a limitation of the MyPy type system I do not normally work around. 

Type guards in TypeScript solve this exact problem.